addons/website_sale/controllers/main.py

```diff
@@ -318,7 +318,7 @@ def shop(self, page=0, category=None, search='', min_price=0.0, max_price=0.0, p
 
         filter_by_price_enabled = website.is_view_active('website_sale.filter_products_price')
         if filter_by_price_enabled:
-            company_currency = website.company_id.currency_id
+            company_currency = website.company_id.sudo().currency_id
             conversion_rate = request.env['res.currency']._get_conversion_rate(
                 company_currency, pricelist.currency_id, request.website.company_id, fields.Date.today())
```

Comment:
Is there no way for the rate computation to fail here as well ? Because the company is not sudoed here either :thinking:

Reply:
I have tested with this commit, there are no issue. But maybe you should investigate more.